src/main/java/com/dropbox/core/android/AuthActivity.java

```diff
@@ -529,10 +617,35 @@ protected void onNewIntent(Intent intent) {
             }
 
             // Successful auth.
-            newResult = new Intent();
-            newResult.putExtra(EXTRA_ACCESS_TOKEN, token);
-            newResult.putExtra(EXTRA_ACCESS_SECRET, secret);
-            newResult.putExtra(EXTRA_UID, uid);
+            if (token.equals(TokenType.OAUTH2.toString())) {
```

Comment:
Actually it should equals. Since the format in migration doc `oauth_tokens=oauth2code:&oauth_token_secret=<code>` is the Uri query string.

With `token = uri.getQueryParameter("oauth_token");` we already fetched `oauth_tokens=oauth2code:` part out of the whole strong.